apiserver/uniter/uniter.go

```diff
@@ -433,7 +433,9 @@ func (u *UniterAPIV3) GetPrincipal(args params.Entities) (params.StringBoolResul
 			if err == nil {
 				principal, ok := unit.PrincipalName()
 				if principal != "" {
-					result.Results[i].Result = names.NewUnitTag(principal).String()
```

Comment:
I'm not sure if an older version of Juju was relying on the value, but I see no strong reason to change away from using the tag format. Can you please leave this method as it was, and just change the API client code?